src/ehrdata/_feature_types.py

```diff
@@ -0,0 +1,326 @@
+from __future__ import annotations
+
+from functools import wraps
+from typing import TYPE_CHECKING, Literal
+
+import numpy as np
+import pandas as pd
+from dateutil.parser import isoparse  # type: ignore
+from lamin_utils import logger
+from rich import print
+from rich.tree import Tree
+from scipy.sparse import issparse
+
+from ehrdata.core.constants import CATEGORICAL_TAG, DATE_TAG, FEATURE_TYPE_KEY, MISSING_VALUES, NUMERIC_TAG
+
+if TYPE_CHECKING:
+    from collections.abc import Iterable
+
+    from ehrdata import EHRData
+
+
+def _detect_feature_type(col: pd.Series) -> tuple[Literal["date", "categorical", "numeric"], bool]:
+    """Detect the feature type of a column in a pandas DataFrame.
```

Comment:
```suggestion
    """Detect the feature type of a :class:`~pandas.Series`.
```

Double typing it is dumb but a column of a DF is a Series. You can add more details on the detected types and how it works in the docstring if you so desire.